### ISSUE TITLE
A lot of "Attempt to prime an already primed trigger for cbValueChange!" messages

### ISSUE THREAD

**Issue by mciepluc** [potentialventures/cocotb#729]
Seems one of the latest merges caused additional production of a lot of such log messages:
```
Attempt to prime an already primed trigger for cbValueChange!
We seem to already be registered, deregistering cbValueChange!
```
I don't quite have time to look into it now, maybe someone know what is going on?

**Comment by themperek** [comment]
@eric-wieser

**Comment by eric-wieser** [comment]
It sounds like the trigger is being primed, but the scheduler never actually holds onto it.

I think calling unprime in __del__ will solve the problem

**Comment by stuarthodgson** [comment]
Why was the change made that changed this behaviour?

**Comment by eric-wieser** [comment]
To remove reference cycles in python objects that made the memory unreclaimable

**Comment by stuarthodgson** [comment]
So there should only have been three objects per signal that were persistent.

**Comment by eric-wieser** [comment]
That should still be true, with the caveat that they now only persist while being used.

I think something very weird is happening here - I don't see the problem on my machine running the tests, and it looks like the trigger is kept alive by the simulator anyway. It would be great if @mciepluc could post an example that causes it.

**Comment by eric-wieser** [comment]
Ok, confirmed in https://api.travis-ci.org/v3/job/469643596/log.txt

**Comment by eric-wieser** [comment]
Mystery solved, will make a patch tonight

**Comment by mciepluc** [comment]
Thanks @eric-wieser for very quick resolution! I can confirm that both #733 and #734 resolve the problem.

**Comment by eric-wieser** [comment]
Well, I broke it so felt a quick fix was necessary!

**Comment by imphil** [comment]
Should be fixed now with #733 merged. @mciepluc please let me know if the current master works for you.

**Comment by mciepluc** [comment]
Thanks all, issue fixed in current master
